core/controller/src/main/scala/whisk/core/entitlement/Entitlement.scala

```diff
@@ -358,10 +367,42 @@ protected[core] abstract class EntitlementProvider(config: WhiskConfig, loadBala
   private def checkThrottleOverload(throttle: Future[RateLimit], user: Identity)(
     implicit transid: TransactionId): Future[Unit] = {
     throttle.flatMap { limit =>
+      val userId = user.authkey.uuid
       if (limit.ok) {
+
+        if (userEvents) {
+          limit match {
+            case c: ConcurrentRateLimit => {
+              val metric =
+                Metric("concurrent_activations", c.count + 1)
+              EventMessage.send(
+                eventProducer,
+                EventMessage(
+                  s"controller$controllerInstance",
+                  metric,
+                  user.subject,
+                  user.namespace.toString,
+                  userId,
+                  metric.getClass.getSimpleName))
```

Comment:
Please get rid of `getClass.getSimpleName` everywhere possible. It's not the cheapest call to make.